packages/wear-os/README.md

```diff
@@ -1,12 +1,428 @@
+| [`startDetecting{prefix}Watch{sensor}Changes`](#start-data-collection-for-a-sensor-with-specific-configuration) | Allows to start the data collection for a `sensor` with a specific configuration (see below). The `prefix` can be used to distinguish among different configurations. |
+| [`stopDetectingWatch{sensor}Changes`](#stop-data-collection-for-a-sensor)                                       | The complement to the previous task. Allows to stop collecting data from `sensor`.                                                                                    |
+| [`sendPlainMessageToWatch`](#send-a-message-to-the-paired-watch)                                                | Allows to send a string-base message to the paired smartwatch. An usage example could be to send information for updating the UI.                                     |
+| [`sendPlainMessageToWatchAndAwaitResponse`](#send-a-message-to-the-paired-watch-and-wait-for-a-response)        | Allows to send a string-base message to the paired smartwatch and to wait for a response from it.                                                                     |
+
+#### Start data collection for a sensor with specific configuration
+To register these tasks for their use, you just need to import them and call their generator functions inside your application's task list:
+
+```typescript
+import { Task } from '@awarns/core/tasks';
+import {
+  startDetectingWatchSensorChangesTask,
+  WatchSensor,
+  WatchSensorDelay,
+} from '@awarns/wear-os';
+
+export const demoTasks: Array<Task> = [
+  startDetectingWatchSensorChangesTask(WatchSensor.ACCELEROMETER, { sensorDelay: WatchSensorDelay.NORMAL, batchSize: 50 }),
+  // startDetectingWatchAccelerometerChanges
+
+  startDetectingWatchSensorChangesTask(WatchSensor.ACCELEROMETER, { sensorDelay: WatchSensorDelay.FASTEST, batchSize: 50 }, 'Fast'),
+  // startDetectingFastWatchAccelerometerChanges
+
+  startDetectingWatchSensorChangesTask(WatchSensor.HEART_RATE, { sensorDelay: WatchSensorDelay.NORMAL, batchSize: 5 }),
+  // startDetectingWatchHeartRateChanges
+
+  startDetectingWatchSensorChangesTask(WatchSensor.GEOLOCATION, { sensorDelay: 5000, batchSize: 5 }),
+  // startDetectingWatchGeolocationChanges
+]
+```
+
+> **Warning**: the data collection for a `WatchSensor` can only be started once, if `startDetectingFastWatchAccelerometerChanges` is executed after
+> `startDetectingWatchAccelerometerChanges` and while the collection is in progress, `startDetectingFastWatchAccelerometerChanges` will be ignored.
+> Therefore, if you want to dynamically change the collection's configuration while the collection is in progress, you will have to stop the collection
+> and start it again with the new desired configuration.
```

Comment:
I didn't ask during the review of the phone-sensors module but, I'm just wondering. Have you tried to run a stop task and a start task triggered by the same event? I mean, imagine that you have a task collecting accelerometer data at a regular rate. Then, an event occurs and I want to stop the collection at that rate and start the collection at a faster rate. Is the order of the execution of the tasks guaranteed by the order in which they appear in the task graph definition? It should, by now I have the doubt...

Reply:
Even if the order of the tasks is guaranteed, the underlying API for the communication doesn't guarantees the order. I mean, if you send a stop/start very fast, it's possible to receive in the watch a start/stop...